StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift

```diff
@@ -530,18 +530,22 @@ extension PaymentSheetFormFactory {
         return StaticElement(view: label)
     }
 
-    func makeContactInformation(includeName: Bool, includeEmail: Bool, includePhone: Bool) -> SectionElement? {
-        let nameElement = includeName ? makeName() : nil
-        let emailElement = includeEmail ? makeEmail() : nil
-        let phoneElement = includePhone ? makePhone() : nil
-
-        let allElements: [Element?] = [nameElement, emailElement, phoneElement]
-        let elements = allElements.compactMap { $0 }
-
+    /// This method returns a "Contact information" Section containing a name, email, and phone field depending on the `PaymentSheet.Configuration.billingDetailsCollectionConfiguration`and your payment method's required fields.
+    /// - Parameter nameRequiredByPaymentMethod: Whether your payment method requires the name field.
+    /// - Parameter emailRequiredByPaymentMethod: Whether your payment method requires the email field.
+    /// - Parameter phoneRequiredByPaymentMethod: Whether your payment method requires the phone field.
+    func makeContactInformationSection(nameRequiredByPaymentMethod: Bool, emailRequiredByPaymentMethod: Bool, phoneRequiredByPaymentMethod: Bool) -> SectionElement? {
+        let nameElement = configuration.billingDetailsCollectionConfiguration.name == .always
+            || (configuration.billingDetailsCollectionConfiguration.name == .automatic && nameRequiredByPaymentMethod) ? makeName() : nil
+        let emailElement = configuration.billingDetailsCollectionConfiguration.email == .always
+            || (configuration.billingDetailsCollectionConfiguration.email == .automatic && emailRequiredByPaymentMethod) ? makeEmail() : nil
+        let phoneElement = configuration.billingDetailsCollectionConfiguration.phone == .always
+            || (configuration.billingDetailsCollectionConfiguration.phone == .automatic && phoneRequiredByPaymentMethod) ? makePhone() : nil
+        let elements = ([nameElement, emailElement, phoneElement] as [Element?]).compactMap { $0 }
```

Comment:
The code seemed cleaner before.  Is there anything we can do to make this simpler?

Reply:
The code before was doing something different - it didn't look at `configuration.billingDetailsCollectionConfiguration` at all.  Now, it handles that logic.

I think this is pretty simple, but it is wordy.  I can try giving `configuration.billingDetailsCollectionConfiguration` a shorter name to make it easier on the eyes.